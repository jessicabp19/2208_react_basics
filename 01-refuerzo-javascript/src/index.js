//Here you can use the code in every file at /basis