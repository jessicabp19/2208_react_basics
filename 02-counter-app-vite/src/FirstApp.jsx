//import { Fragment} from 'react'
import PropTypes from 'prop-types';

//Async func no se renderiza directamente por que devuelve un obj(promesa)
const getResult = () => { 
    return "Hola! Mundo!"
} 
const obj = {
    messaje: 'Hola mundo',
    title: 'Titulo'
};

export const FirstApp = ( {title, subtitle} ) => { 
    
    //console.log(props);

    return (
    // <Fragment>
    //     <h1>Jessica</h1>
    //     <p>Soy un parrafo</p>    
    // </Fragment>


    <>
        <h1 data-testid="test-title">{title}</h1>
        <h2>{subtitle}</h2>
        {/* <h2>{subtitle}</h2> */}
        {/* <code>{JSON.stringify(obj)}</code> */}
        {/* <p>{getResult()}</p>     */}
    </>
  )
}

FirstApp.propTypes = {
    title: PropTypes.string.isRequired,
    subtitle: PropTypes.string.isRequired
}

FirstApp.defaultProps = {
    name: 'Jessica',
    //title: 'No hay titulo',
    subtitle: 'No hay subtitulo'
}