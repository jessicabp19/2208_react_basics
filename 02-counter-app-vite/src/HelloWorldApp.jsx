export function App(){
    return (<h1>Hello Mundo</h1>);
}

//Shorcut: rafc (2 opciones)

export const HelloWorldApp = () => {
  return (
    <div>HelloWorldApp</div>
  )
}
