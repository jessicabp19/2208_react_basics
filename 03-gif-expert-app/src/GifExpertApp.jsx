
import { useState } from "react";
import { AddCategory, GifGrid } from "./components";

export const GifExpertApp = () => {

/* Cuando querramos almacenar información y esa información tiene que cambiar el HTML
Usualmente lo primero que deberíamos pensar: Ok, necesito algún Hook de React para mantener el estado */

  const [categories, setCategories] = useState([ "Grey's Anatomy" ]);

  /* 
  - Los hooks no tienen nombre como tal, se enumeran
  - Nunca poner algún Hook the react dentro de un condicional 
  */

  const onAddCategory = (newCategory) => {

    if ( categories.includes(newCategory) ) return;

    // categories.push(newCategory) // Esto No :)
    setCategories( [newCategory, ...categories] ); //Operador Spread
    //setCategories( cat => [...cat, 'PopCorn'] ); //Callback
  }

  return (
    <> 
      <h1>GifExpertApp</h1>
      <AddCategory
        onNewCategory={ event => onAddCategory(event) } 
      />

      {
        
      categories.map( (category) => (
        <GifGrid 
          key={category} 
          category={category} 
        />
      ))
      }
    </>
  )
}
