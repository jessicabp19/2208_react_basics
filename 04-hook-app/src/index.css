body {
    padding: 20px;
}

button {
    margin-right: 10px;
}

input {
    background-color: white;
    border-radius: 5px;
    border: 1px solid rgb(97, 32, 158);
    color: black;
    font-size: 1.2rem;
    outline: none;
    padding: 10px 15px;
    width: 100%;
    margin-top: 20px;
    margin-bottom: 30px;
  }