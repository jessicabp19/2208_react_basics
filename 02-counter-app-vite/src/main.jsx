import React from 'react'
import ReactDOM from 'react-dom/client';
import { CounterApp } from './CounterApp';
import { FirstApp } from './FirstApp';
import { App, HelloWorldApp } from './HelloWorldApp';

import './styles.css';

ReactDOM.createRoot( document.getElementById('root')).render(
    <React.StrictMode>
        {/* <App /> */}
        {/* <HelloWorldApp /> */}

        {/* <FirstApp title="Hola! Soy yo" subtitle={ 123 } /> */}
    
        <CounterApp value={100} />
    </React.StrictMode>
);
