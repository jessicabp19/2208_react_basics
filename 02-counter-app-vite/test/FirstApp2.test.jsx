import { render, screen } from '@testing-library/react'; //screen es la rep de nuestro comp rend
import { FirstApp } from "../src/FirstApp";

describe('Tests in <FirstApp />', () => {

    const title     = 'Hola, Soy Jessica';
    const subtitle  = 'Soy un subtitulo';

    test('should match the snapshot', () => {
        
        const { container } = render( <FirstApp title={title} /> );
        expect( container ).toMatchSnapshot(); 

    });
    
    test('should show the msj "Hola, Soy Jessica"', () => {

        render( <FirstApp title={ title } /> );
        expect( screen.getByText(title) ).toBeTruthy();
        // screen.debug();

    });
    
    test('should show the title in h1 tag', () => { 
        
        render( <FirstApp title={title} /> );
        expect( screen.getByRole('heading', { level:1 }).innerHTML ).toContain( title ) ;

     });

     test('should show the subtitle sent by props', () => { 
        
        render( 
            <FirstApp title={ title } subtitle={subtitle} /> 
        );
        expect( screen.getAllByText( subtitle ).length ).toBe(1);//2
      })

})
