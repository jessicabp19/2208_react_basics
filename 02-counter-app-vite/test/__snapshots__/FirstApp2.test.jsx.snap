// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <FirstApp /> should match the snapshot 1`] = `
<div>
  <h1
    data-testid="test-title"
  >
    Hola, Soy Jessica
  </h1>
  <h2>
    No hay subtitulo
  </h2>
</div>
`;
